pkgs/tools/archivers/p7zip/default.nix

```diff
@@ -42,7 +42,8 @@ stdenv.mkDerivation rec {
 
   setupHook = ./setup-hook.sh;
 
-  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.cc.isClang "-Wno-error=c++11-narrowing";
+  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.cc.isClang "-Wno-error=c++11-narrowing"
+    + "-Wno-error=narrowing";
```

Comment:
https://sourceforge.net/p/p7zip/bugs/226/

(Also, this will break on clang, the flags need to be space-separated)